lib/chef/provider/windows_task.rb

```diff
@@ -201,308 +224,303 @@ def action_disable
+          new_resource.start_time = Time.now.strftime("%H:%M")
         end
-        Chef::Log.debug("running: ")
-        Chef::Log.debug("    #{cmd}")
-        shell_out!(cmd, returns: [0])
       end
-      # rubocop:enable Style/StringLiteralsInInterpolation
 
-      def task_need_update?
-        return true if (new_resource.command &&
-            current_resource.command != new_resource.command.tr("'", '"')) ||
-            current_resource.user != new_resource.user ||
-            current_resource.run_level != new_resource.run_level ||
-            current_resource.cwd != new_resource.cwd ||
-            current_resource.frequency_modifier != new_resource.frequency_modifier ||
-            current_resource.frequency != new_resource.frequency ||
-            current_resource.idle_time != new_resource.idle_time ||
-            random_delay_updated? || execution_time_limit_updated? ||
-            (new_resource.start_day && new_resource.start_day != "N/A" && start_day_updated?) ||
-            (new_resource.start_time && new_resource.start_time != "N/A" && start_time_updated?)
-        begin
-          return true if new_resource.day.to_s.casecmp(current_resource.day.to_s) != 0 ||
-              new_resource.months.to_s.casecmp(current_resource.months.to_s) != 0
-        rescue
-          Chef::Log.debug "caught a raise in task_needs_update?"
+      def update_task(task)
+        converge_by("#{new_resource} task updated") do
+          task.application_name = new_resource.command
+          task.trigger = trigger
+          # TODO: Check update for execution time limit
+          task.max_run_time = new_resource.execution_time_limit.to_i * 1000
+          task.configure_settings(config_settings)
+          task.configure_principals(principal_settings)
         end
-
-        false
       end
 
-      # Comparing random_delay values using ISO8601::Duration object Ref: https://github.com/arnau/ISO8601/blob/master/lib/iso8601/duration.rb#L18-L23
-      # di = ISO8601::Duration.new(65707200)
-      # ds = ISO8601::Duration.new('P65707200S')
-      # dp = ISO8601::Duration.new('P2Y1MT2H')
-      # di == dp # => true
-      # di == ds # => true
-      def random_delay_updated?
-        if new_resource.random_delay.nil?
-          false
-        elsif current_resource.random_delay.nil? && new_resource.random_delay == "PT0S" # when user sets random_dealy to 0 sec
-          false
-        elsif current_resource.random_delay.nil?
-          true
-        else
-          ISO8601::Duration.new(current_resource.random_delay) != ISO8601::Duration.new(new_resource.random_delay)
+      def trigger
+        date = new_resource.start_day.split('/') if new_resource.start_day
+        time = new_resource.start_time.split(':') if new_resource.start_time
+        #TODO currently end_month, end_year and end_year needs to be set to 0. If not set win32-taskscheduler throwing nil into integer error.
+        trigger_hash = {}
```

Comment:
need not to be declared, we are defining it in next line,
Declaration is not necessary